tiledb/common/memory_tracker.h

```diff
@@ -115,6 +115,7 @@ enum class MemoryType {
   ATTRIBUTES,
   DIMENSION_LABELS,
   DIMENSIONS,
+  DOMAINS,
```

Comment:
Ope! I missed that you missed the you forgot the `memory_type_to_str` switch case.